lib/ui/src/components/preview/tools/copy.tsx

```diff
@@ -5,15 +5,18 @@ import { getStoryHref, IconButton, Icons } from '@storybook/components';
 import { Consumer, Combo } from '@storybook/api';
 import { Addon } from '@storybook/addons';
 
-const { PREVIEW_URL } = global;
+const { PREVIEW_URL, document } = global;
 
 const copyMapper = ({ state }: Combo) => {
   const { storyId, refId, refs } = state;
+  const { location } = document;
   const ref = refs[refId];
 
   return {
     refId,
-    baseUrl: ref ? `${ref.url}/iframe.html` : (PREVIEW_URL as string) || 'iframe.html',
+    baseUrl: ref
+      ? `${ref.url}/iframe.html`
+      : (PREVIEW_URL as string) || `${location.origin}/iframe.html`,
```

Comment:
`location.origin` doesn't work for non-root storybooks. should this be `location.href`? @soso-15315

Reply:
Thanks for the comment.
I had overlooked non-root storybooks.

> should this be location.href?

Since the URL will be merged later, I thought it would be better to use a different method.

https://github.com/storybookjs/storybook/blob/next/lib/ui/src/components/preview/tools/copy.tsx#L31-L33

I modified it as follows to get the complete path.

```ts
let baseUrl = `${location.origin}${location.pathname}`;
if (!baseUrl.endsWith('/')) baseUrl += '/';

... || `${baseUrl}iframe.html`
```

What do you think?